hosts/azure/jenkins-controller/configuration.nix

```diff
@@ -318,10 +295,15 @@ in {
     post-build-hook = ${post-build-hook}
   '';
 
-  # TODO: use https://caddyserver.com/docs/caddyfile-tutorial#environment-variables for domain
+  services.rclone-http = {
+    enable = true;
+    protocol = "webdav";
+    remote = ":azureblob:binary-cache-v1";
+  };
+
   services.caddy = {
     enable = true;
-    configFile = pkgs.writeTextDir "Caddyfile" ''
+    configFile = pkgs.writeText "Caddyfile" ''
```

Comment:
Why do we change the `writeTextDir` to `writeText`?

Reply:
See the discussion in https://github.com/NixOS/nixpkgs/issues/272532.

Peeking at `nixos/tests/caddy.nix`, setting it like this should work?!

Reply:
this is the new recommended way.